test/test_instructions/test_create_instructions.py

```
@@ -42,17 +91,41 @@ def test_dtw_to_stitches_case_failed(dtw: np.ndarray) -> None:
+    dtw = np.array([[0, 0], [1, 1], [2, 2]])
+    with pytest.raises(ValueError, match="creases must be a vector"):
+        dtw_to_stitches(dtw, creases)
+
+
+@pytest.mark.parametrize('creases', [
+    np.array([0, 0]),        # one element too short
+    np.array([0, 0, 0, 0]),  # one element too long
+])
+def test_dtw_to_stitches_creases_wrong_length(creases):
+    """Tests that dtw_to_stitches raises ValueError when creases length doesn't match the earlier row."""
+    dtw = np.array([[0, 0], [1, 1], [2, 2]])  # earlier row has 3 vertices (indices 0–2)
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)